#!/usr/bin/env python
PACKAGE = "pointcloud_filter"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

size_enum = gen.enum([ gen.const("statistical_outiler_removal_filter", int_t, 0, "statistical outiler removal filter"),
                       gen.const("radius_outlier_removal_filter", int_t, 1, "radius outlier removal filter")],
                     "Kind of filter")

gen.add("outlier_filter", int_t, 0, "Select kind of filter", 1, 0, 1, edit_method=size_enum)
gen.add("sor_k", double_t, 0, "sor_k",    25, 0,   100)
gen.add("sor_mul_thresh", double_t, 0, "sor_mul_thresh",    1, 0,   20)
gen.add("ror_rs", double_t, 0, "ror_rs",    0.1, 0,   2)
gen.add("ror_min_neighbors", double_t, 0, "ror_min_neighbors",    14, 0,   20)

exit(gen.generate(PACKAGE, "pointcloud_filter", "OutlierFilter"))
